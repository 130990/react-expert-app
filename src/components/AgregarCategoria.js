import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AgregarCategoria = ({ setCategorias }) => {
    const [inputValue, setInputValue] = useState('');
    const handleCambiarInput = (e) => {        
        setInputValue(e.target.value)
    }
    const handlEnviar = (e) => {
        e.preventDefault();
        console.log('Llamado Submit......');
        if (inputValue.length > 2) {
            setCategorias(cate => [inputValue,...cate])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handlEnviar}>
            <input type="text" value={inputValue} onChange={handleCambiarInput} />
        </form>
    )
}

AgregarCategoria.propTypes ={
    setCategorias: PropTypes.func.isRequired
}
