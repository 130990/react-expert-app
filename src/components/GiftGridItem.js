import React from 'react'
import '../index.css';
import PropTypes from 'prop-types'

export const GiftGridItem = ({title, url }) => {
    return (
        <div className="card animate__animated animate__fadein">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes ={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
