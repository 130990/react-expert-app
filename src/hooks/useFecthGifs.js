import { useState, useEffect } from 'react'
import { obtenerGifts } from '../helpers/getGifs';

export const useFecthGifs = (categoria) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        obtenerGifts(categoria).then(imgs => setState({
            data: imgs,
            loading: false
        }));
    },
        [categoria]
    ); // con la lista de dependensias vacias seria como el DidMount(), en react basado en clases
    //cuando s epasa la categoria por depencia, dispara a que se recargue solo en caso de q la categoria haya cargado

    return state;
}
