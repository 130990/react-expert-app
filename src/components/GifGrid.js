import React from 'react'
import { useFecthGifs } from '../hooks/useFecthGifs'
import { GiftGridItem } from './GiftGridItem'

export const GifGrid = ({ categoria }) => {

    const { data: imagenes, loading } = useFecthGifs(categoria);


    return (
        <>
            <h3>{categoria}</h3>
            {loading &&<p className='animate__animated animate__flash'>Cargando imagenes...</p> }
            <div className="card-grid" >
                {
                    imagenes.map((img) => (
                        <GiftGridItem key={img.id} {...img} /> //enviando las imagenes con el spread, permite enviar las propiedades como un solo objeto, q puede desestructurase del otro lado
                    ))
                }
            </div>
        </>
    )
}
