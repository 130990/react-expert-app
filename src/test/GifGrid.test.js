import React from 'react'
import { shallow } from 'enzyme';
import { GifGrid } from '../components/GifGrid'
import '@testing-library/jest-dom'
import { useFecthGifs } from '../hooks/useFecthGifs';
jest.mock('../hooks/useFecthGifs');

describe('Pruebas de GifGrid', () => {
    test('Debe de ejecutar bien el componente <GifGrid /> ', () => {
        const categoria = 'Goku';
        useFecthGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const ResultadoObtenido = shallow(<GifGrid categoria={categoria} />)
        expect(ResultadoObtenido).toMatchSnapshot();
    })

    test('Debe de mostrar items cuando se cargan imagenes del useFecthGifs', () => {
        const gifs =[{
            url:'https:///pruebab/a.png',
            title: 'Prueba',
            id: 123
        }];
        useFecthGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const categoria = 'Goku';
        const ResultadoObtenido = shallow(<GifGrid categoria={categoria} />)
        //expect(ResultadoObtenido).toMatchSnapshot();
        expect(ResultadoObtenido.find('p').exists()).toBe(false);
        expect(ResultadoObtenido.find('GiftGridItem').length).toBe(gifs.length);
    })


})
