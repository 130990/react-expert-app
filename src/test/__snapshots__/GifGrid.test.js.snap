// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGrid Debe de ejecutar bien el componente <GifGrid />  1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando imagenes...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
